test/encode.cpp

```diff
@@ -15,23 +15,26 @@ using namespace quicr::messages;
 
 TEST_CASE("MessageBuffer Swap Bytes")
 {
-  uint16_t u16 = 0x1234u;
-  uint32_t u32 = 0x12345678u;
-  uint64_t u64 = 0x123456789ABCDEF0u;
-  Name u128 = 0x123456789ABCDEF0123456789ABCDEF0_name;
-
-  CHECK_NE(u16, swap_bytes(u16));
-  CHECK_NE(u32, swap_bytes(u32));
-  CHECK_NE(u64, swap_bytes(u64));
-  CHECK_NE(u128, swap_bytes(u128));
+  const auto u16 = uint16_t(0x1234U);
+  const auto u32 = uint32_t(0x12345678U);
+  const auto u64 = uint64_t(0x123456789ABCDEF0U);
+  const auto u128 = 0x123456789ABCDEF0123456789ABCDEF0_name;
+
+  // XXX(richbarn): This test will return incorrect results on a big-endian
+  // system, since swap_bytes actually converts to big-endian.
+  CHECK(u16 != swap_bytes(u16));
+  CHECK(u32 != swap_bytes(u32));
+  CHECK(u64 != swap_bytes(u64));
+  CHECK(u128 != swap_bytes(u128));
```

Comment:
```suggestion
  if constexpr (std::endian::native == std::endian::little) {
    CHECK(u16 != swap_bytes(u16));
    CHECK(u32 != swap_bytes(u32));
    CHECK(u64 != swap_bytes(u64));
    CHECK(u128 != swap_bytes(u128));
  }
```

Reply:
What's wrong with `CHECK_NE`?

Reply:
I think there was something internal to it that clang-tidy was unhappy with, but I also just think that `_EQ` / `_NE` suffixes are less clear than just writing something like you would in an `if` statement.  Note that doctest rewrites the equality operators so that they do the right thing for it.

Reply:
Actually, I just went ahead and fixed the test.  I created byte-swapped values for the little-endian case, and compared the output of `swap_bytes` to the correct value for each endianness.